### PR TITLE
feat(j-s): Indictment Conclusion

### PR DESCRIPTION
**Pull request by gudjong**
# Indictment Conclusion

[Hreinsa til í skjölum sem fylgja lyktum í ákærumálum](https://app.asana.com/0/1199153462262248/1207418628462283/f)
[Endurúthlutun - framhald málsmeðferðar](https://app.asana.com/0/1199153462262248/1207539443130904/f)

## What

- Removes the main hearing state and redistribute transaction.
- Rewrites the Conslusion screen adding a separate option for scheduling a new court date and simplifying case updates and screen initialization.
- Removes redundant ruling files when an indictment is completed with a fine or cancellation.
- Implements validation for the conclusion step in the form stepper.
- Rewrites info card for scheduled indictment cases.
- Always disables the arraignment date screen after an indictment decision has been made.
- Fixes an issue with validating the arraignment date step in the form stepper.
- Fixes a few issues with case tags.
- Migrates all cover letter case files to simple case files (leftover from another PR).

## Why

- In preparation for main hearing implementation.

### Conclusion page when postponing
image

### Conclusion page when scheduling new court date
image

### Conclusion page when postponing until verdict
image

### Conclusion page when comleting
image

### Conclusion page when redistributing
image

### Arraignment date in case list
image

### Postponement in case list
image

### New court date in case list
image

### Postponement until verdict in case list
image

### Completing in case list
image

### Redistributing in case list
image

### Info card for arraignment date
image

### Info card for postponement
image

### Info card for new court date
image

### Info card for postponement until verdict
image

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
  - Added `indictmentDecision` field to various components and services to enhance case handling logic.
  - Introduced new message titles and descriptions for scheduled indictment cases and various case states.

- **Bug Fixes**
  - Removed `CaseState.MAIN_HEARING` from access and query filters for better consistency.
  - Eliminated `REDISTRIBUTE` transition to simplify case transitions.

- **Refactor**
  - Updated multiple components and services to handle `indictmentDecision` conditions.
  - Refined validation logic in `validate.ts` to match updated case states and decisions.

- **Tests**
  - Updated test cases to reflect removal of `CaseState.MAIN_HEARING` and adjustments in case handling logic.

### CODE REVIEW

File: apps/judicial-system/web/src/components/InfoCard/InfoCardCaseScheduledIndictment.tsx [hunk 1]
```diff
@@ -1,33 +1,67 @@
...
+              <Text marginTop={2}>
+                {formatMessage(strings.schedulingUntilVerdictText)}
+              </Text>
+            ),
+          },
+        ]}
+        icon="calendar"
+      />
+    ) : (
+      // indictmentDecision === IndictmentDecision.SCHEDULING
```
Comment (unakb) [inline apps/judicial-system/web/src/components/InfoCard/InfoCardCaseScheduledIndictment.tsx:57]: Is this supposed to be here?

File: apps/judicial-system/web/src/components/InfoCard/InfoCardCaseScheduledIndictment.tsx [hunk 2]
```diff
@@ -1,33 +1,67 @@
...
+      <InfoCardCaseScheduled
+        court={court}
+        courtDate={courtDate}
+        courtRoom={courtRoom}
+        title={formatMessage(strings.schedulingTitle)}
+      />
+    )
```
[the remark this answers is not in the document]
Comment (gudjong) [reply apps/judicial-system/web/src/components/InfoCard/InfoCardCaseScheduledIndictment.tsx:64]: Not all values are relevant here, but the plan is to rewrite this component in the near future. I will not do it now.

File: apps/judicial-system/web/src/utils/validate.ts [hunk 3]
```diff
@@ -446,8 +448,39 @@ export const isDefenderStepValid = (workingCase: Case): boolean => {
...
+          )
+        default:
+          return false
+      }
+    case IndictmentDecision.POSTPONING_UNTIL_VERDICT:
+    case IndictmentDecision.REDISTRIBUTING:
+      return true
+    default:
+      return false
+  }
```
[the remark this answers is not in the document]
Comment (gudjong) [reply apps/judicial-system/web/src/utils/validate.ts:483]: Good idea. I will make this change.

**Review by unakb — APPROVED** [review]
Nice, just one question from me otherwise looks good

File: apps/judicial-system/web/src/components/InfoCard/InfoCardCaseScheduledIndictment.tsx (hunk 1 above)
Comment (gudjong) [reply apps/judicial-system/web/src/components/InfoCard/InfoCardCaseScheduledIndictment.tsx:57]: Yes, it tells us what value we are expecting. I want this component rewritten at a later date to make it clearer.

File: apps/judicial-system/web/src/utils/validate.ts [hunk 4]
```diff
@@ -421,7 +423,7 @@ export const isSubpoenaStepValid = (
         courtDate ?? workingCase.arraignmentDate?.date,
         ['empty', 'date-format'],
       ],
-      [courtRoom, ['empty']],
+      [courtRoom ?? workingCase.arraignmentDate?.location, ['empty']],
```
[the remark this answers is not in the document]
Comment (gudjong) [reply apps/judicial-system/web/src/utils/validate.ts:426]: We want undefined or null.

**Pull request merged**